src/pages/QuizPage/QuizPage.jsx

```diff
@@ -0,0 +1,127 @@
+import {useEffect, useState} from 'react'
+import { useQuiz } from '../../context/quiz-context'
+import { useNavigate, useParams } from 'react-router-dom';
+import Navbar from '../../components/Navbar/Navbar';
+import './QuizPage.css'
+const QuizPage = () => {
+
+  const {getQuizById, selectedQuizId, questions} = useQuiz();
+  const { id } = useParams();
+  const [currentQuestion, setCurrentQuestion] = useState(0);
+  const [showScore, setShowScore] = useState(false);
+  const [score, setScore] = useState(0);
+  const [userAns, setUserAns] = useState("")
+  const [showRules, setShowRules] = useState(true);
```

Comment:
here usestate hook count is 5 so we can use reducer, even i did same so will try to use reducer if count goes beyond 3